src/timing.c

```diff
@@ -60,11 +62,22 @@ void jl_print_timings(void)
         root_time -= jl_timing_counts[i];
     }
     jl_timing_counts[0] = root_time;
+    fprintf(stderr, "\nJULIA TIMINGS\n");
     for (int i = 0; i < JL_TIMING_LAST; i++) {
         if (jl_timing_counts[i] != 0)
             fprintf(stderr, "%-25s : %5.2f %%   %" PRIu64 "\n", jl_timing_names[i],
                     100 * (((double)jl_timing_counts[i]) / total_time), jl_timing_counts[i]);
     }
+
+    fprintf(stderr, "\nJULIA COUNTERS\n");
+    uint64_t val = 0;
+#ifdef USE_TIMING_COUNTS
+#define X(name) val = jl_atomic_load_relaxed(&jl_timing_counters[(int)JL_TIMING_COUNTER_##name].basic_counter); \
+    if (val != 0) \
+        fprintf(stderr, "%-25s : %" PRIu64 "\n", #name, val);
```

Comment:
It looks like this `fprintf` will never happen, right?

Also, can we either wrap the entire function body in `#ifdef USE_TIMING_COUNTS`, or leave the guard out? It's unusual to guard only half of the output

Reply:
We set val in the line above, but yeah I can change to int64 and guard the whole function.

Reply:
Ah I missed the first `\` and thought this line wasn't in the macro.

Logic looks good to me in that case